tests/console/yast2_lan_hostname_dhcp_yes_eth.pm

```diff
@@ -0,0 +1,42 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Package: yast2-network
+# Summary: Verify that correct value is stored in network config when
+# setting hostname via DHCP to 'yes: <ethernet network interface>' in
+# YaST2 lan module (https://bugzilla.suse.com/show_bug.cgi?id=984890)
+# Maintainer: QA SLE YaST team <qa-sle-yast@suse.de>
+
+use parent 'yast2_lan_hostname_base';
+use strict;
+use warnings;
+use testapi;
+use scheduler qw(get_test_suite_data);
+use YaST::Module;
+
+sub run {
+    my $network_interface = script_output('ls /sys/class/net | grep ^e');
+    my $test_data         = get_test_suite_data();
+
+    YaST::Module::open({module => 'lan', ui => $test_data->{ui}});
+    my $network_settings = $testapi::distri->get_network_settings();
+    $network_settings->set_hostname_via_dhcp({
+            dhcp_option     => "yes: $network_interface",
+            confirm_warning => $test_data->{confirm_warning}
+    });
+    $network_settings->save_changes();
+
+    assert_screen 'console-visible', 180;
```

Comment:
why we should still use needles, can we just change console here?

Reply:
Initially I had `select_console 'root'` here. By the way, it also uses needles inside:)

But I had some issues with waiting and thought it is due to that. So, I returned the approach that was used in the initial `yast2_lan_hostname` test module. The issues were not gone:) So I can revert to `select_console 'root'`, but I have better approach here, that I want to apply. 

I want to create something like `wait_for_close` in YaST::Module and use `wait_serial` inside to check that module is closed.

Reply:
I wonder if we want to promote this method and extract to some method? No need to do it now though.